server/bin/pbench-index.py

```diff
@@ -105,6 +105,17 @@ def main(options, name):
                 - Raises an exception caught be a new, outer-most, try/except
                   block that does not have a finally clause (as you don't want
                   any code execution in the finally block).
+        4. Re-evaluate the list of tar balls to index after indexing of the
+            current tar ball has finished
+            - SIGHUP
+            - Report the current state of indexing
+                - Current tar ball being processed
+                - How many remaining
+                - How many have been completed
+                - Errors encountered
+            - Handler Behavior:
+                1. Kicks a special thread to log indexing state data
```

Comment:
The output isn't being done in a thread; you're generating it inline. Which I think is fine, since this isn't a lot of code. Although I think the original idea was that instead of synchronously looking for an interrupt the next time around the loop at this one spot, the interrupt handler would create a thread that would log the immediate status in parallel, regardless of where in the loop we are. (And note that trying to refresh `tarballs` in that context would be even more problematic than it is here.)

Reply:
I think doing it "in-line" is far better than trying to use a thread:

- Python threads don't actually offer much concurrency -- Python bytecode execution is done under a global lock, so a thread executing "Python code" only executes in parallel with other threads which _aren't_ running "Python code" (like stuff doing I/O or executing an external library such as NumPy which is written in C).
- Signal delivery is apparently also synchronous with respect to bytecode execution, so the notification is going to wait until "a good time", anyway.  (I.e., it might not actually interrupt anything.)
- The in-line approach is _much_ simpler and easier to understand and maintain.
- The execution of the interrupt is not performance-critical.
- Adding threads makes the update of the `tarballs` list much more complicated.

Reply:
oh, I must have missed the comment ;P